Adjusts parser for "show ap cdp neighbor"

jamiereid opened the pull request:
## Description
Resolves #723

Adjust regex to include the ap neighbor IP as it is output on 9800 platform v17.x

## Motivation and Context
The parser fails as it's not able to match when the ap cdp neighbor IP is in the same line as the other information.

Commit:
Adjusts parser for "show ap cdp neighbor"

Resolves #723

Adjust regex to include the ap neighbor IP as it is output on 9800
platform v17.x

ThomasJRyan commented on src/genie/libs/parser/iosxe/show_ap.py:
```diff
@@ -1200,7 +1200,7 @@ def cli(self, output=None):
         neighbor_count_capture = re.compile(r"^Number\s+of\s+neighbors:\s+(?P<neighbor_count>\d+)$")
         # 0221-cap22                   10.8.33.106                              a02-21-sd-sw1.cisco.com TenGigabitEthernet3/0/47
         neighbor_info_capture = re.compile(
```
Add a comment with an example of what's being matched

ThomasJRyan commented on src/genie/libs/parser/iosxe/show_ap.py:
```diff
@@ -1235,6 +1235,12 @@ def cli(self, output=None):
                 ap_cdp_neighbor_dict['ap_name'][ap_name]['neighbor_port'] = neighbor_port
+                if groups['neighbor_ip']:
+                    neighbor_ip = groups['neighbor_ip']
+                    if not ap_cdp_neighbor_dict['ap_name'][ap_name].get('neighbor_ip_addresses', {}):
+                        ap_cdp_neighbor_dict['ap_name'][ap_name]['neighbor_ip_addresses'] = []
+                    ap_cdp_neighbor_dict['ap_name'][ap_name]['neighbor_ip_addresses'].append(neighbor_ip)
```
```suggestion
                    address_list = ap_cdp_neighbor_dict['ap_name'][ap_name].setdefault("neighbor_ip_addresses", [])
                    address_list.append(neighbor_ip)
```

ThomasJRyan reviewed (changes requested):
Thank you for your contribution.

A few things

1) Please include a unittest for this change. You can find existing unittests for this parser [here](https://github.com/CiscoTestAutomation/genieparser/tree/main/src/genie/libs/parser/iosxe/tests/ShowApCdpNeighbor/cli/equal)

2) While I won't hold it against you, since you're updating this parser anyways I would appreciate if you could update line 1152 from `Optional("neighbor_ip_addresses"): list` to `Optional("neighbor_ip_addresses"): ListOf(str)` as it's a bit out of date for what we expect from a parser these days

jamiereid commented:
Thank you @ThomasJRyan, and apologies for the delay in response. I have pushed some additional commits that (hopefully correctly) addresses your feedback. Please let me know :)

ThomasJRyan commented:
Looks like your changes are causing existing unittests to fail.

![image](https://github.com/user-attachments/assets/50755160-c7fe-4148-aceb-aead38381d54)

This will need to be fixed up before we can merge. You can run the unittests locally by going into the `tests` folder and running the [folder_parsing_job.py](https://github.com/CiscoTestAutomation/genieparser/blob/main/tests/folder_parsing_job.py) file

jamiereid commented:
I've looked into these most recent failures a little more but it's not clear to me the fix.

When I run the tests "directly" by changing into the `tests` directory and executing `python folder_parsing_job.py` they complete and all pass. Digging further into the way github is setup to run the tests though, I see it runs them through `pyats` `pyats run job folder_parsing_job.py --no-mail --no-archive`. This is failing locally for me as well, with the same VersionConflict error as github.

```
pkg_resources.VersionConflict: (genie.libs.parser 25.5 (/home/runner/work/genieparser/genieparser/src), Requirement.parse('genie.libs.parser<25.7.0,>=25.6.0'))
```

Are you able to advise, @ThomasJRyan?

Taarini reviewed (changes requested):
Your branch is not up to date with main. Please merge with the latest.

jamiereid commented:
rebased on top of main
